### ISSUE TITLE
[BUG]: mlflow incorrectly logging models "Lasso Least Angle Regression" and "Least Angle Regression"

### ISSUE THREAD

**Issue by TahaSha** [pycaret/pycaret#2811]
### pycaret version checks

- [X] I have checked that this issue has not already been reported [here](https://github.com/pycaret/pycaret/issues).

- [X] I have confirmed this bug exists on the [latest version](https://github.com/pycaret/pycaret/releases) of pycaret.

- [X] I have confirmed this bug exists on the master branch of pycaret (pip install -U git+https://github.com/pycaret/pycaret.git@master).

### Issue Description

mlflow logs the name of both models "Least Angle Regression" and "Lasso Least Angle Regression" as "Least Angle Regression".

When looking into the `get_logs()` you can see both of those models have unique `run_id` but both have the same `tags.mlflow.runName`.

Python Version: 3.9.5
PyCaret Version: '3.0.0.rc3'
Pandas Version: 1.4.3

### Reproducible Example

```python
import pandas as pd
from pycaret.regression import *
from pycaret.datasets import get_data
dataset = get_data('diamond')

EXPERIMENT_NAME = 'diamond_experiment'
s = setup(data=dataset, target='Price', log_experiment=True, experiment_name=EXPERIMENT_NAME, session_id=42, verbose=True)

model = compare_models(verbose=False)

print(f"Notice Least Angle Regression is not unique:\n{get_logs(experiment_name=EXPERIMENT_NAME)['tags.mlflow.runName'].value_counts()}")

# Loop through all models in the `compare_models()` (20 models) function and get the length of the dataframe of that specific model in the logs
# There should be a single unique value for each model
for model in pull().Model.tolist():
    print(f"{model} - {len(get_logs(experiment_name=EXPERIMENT_NAME)[get_logs(experiment_name=EXPERIMENT_NAME)['tags.mlflow.runName'] == model])}")

# Further investigation: model Least Angle Regression has 2 instances (should be Lasso Least Angle Regression and Least Angle Regression)
get_logs(experiment_name=EXPERIMENT_NAME)[get_logs(experiment_name=EXPERIMENT_NAME)['tags.mlflow.runName'] == 'Least Angle Regression']
```
```


### Expected Behavior

`tags.mlflow.runName` parameter from `get_logs()` is unique (given a single experiment) and contains all model names from `compare_models()`

### Actual Results

```python-traceback
When looking into the `tags.mlflow.runName` you can see they are all unique but Least Angle Regression is there twice and Lasso Least Angle Regression isn't there at all. Could this be logged incorrectly?

Gradient Boosting Regressor - 1
CatBoost Regressor - 1
Light Gradient Boosting Machine - 1
Extreme Gradient Boosting - 1
Lasso Regression - 1
Ridge Regression - 1
Linear Regression - 1
Lasso Least Angle Regression - 0
Least Angle Regression - 2
Extra Trees Regressor - 1
Random Forest Regressor - 1
AdaBoost Regressor - 1
Decision Tree Regressor - 1
Orthogonal Matching Pursuit - 1
Elastic Net - 1
Huber Regressor - 1
Bayesian Ridge - 1
K Neighbors Regressor - 1
Dummy Regressor - 1
Passive Aggressive Regressor - 1
```

### Installed Versions

<details>
System:
    python: 3.9.5 (v3.9.5:0a7dcbdb13, May  3 2021, 13:17:02)  [Clang 6.0 (clang-600.0.57)]
executable: PATH_TO_ENV/venv/bin/python
   machine: macOS-10.16-x86_64-i386-64bit

PyCaret required dependencies:
                 pip: 21.1.1
          setuptools: 56.0.0
             pycaret: 3.0.0.rc3
             IPython: 8.4.0
          ipywidgets: 8.0.0rc0
                tqdm: 4.64.0
               numpy: 1.21.6
              pandas: 1.4.3
              jinja2: 3.1.2
               scipy: 1.5.4
              joblib: 1.1.0
             sklearn: 1.1.1
                pyod: Installed but version unavailable
            imblearn: 0.9.1
   category_encoders: 2.5.0
            lightgbm: 3.3.2
               numba: 0.55.2
            requests: 2.28.0
          matplotlib: 3.5.2
          scikitplot: 0.3.7
         yellowbrick: 1.4
              plotly: 5.9.0
             kaleido: 0.2.1
         statsmodels: 0.13.2
              sktime: 0.11.4
               tbats: Installed but version unavailable
            pmdarima: 1.8.5
              psutil: 5.9.1
</details>

**Comment by moezali1** [comment]
@Yard1 Can you give me a hand here? I ended up spending a lot of time in figuring out where is it coming from. The names inside `containers/regression.py` seems to be fine but even then the run name is wrong.

![image](https://user-images.githubusercontent.com/54699234/204138706-db0d0cc3-9a08-46d3-8037-fbaee414876b.png)

Any ideas?
